### ISSUE TITLE
Crash Report

### ISSUE THREAD

**Issue by g3grau** [mu-editor/mu#2045]
mu-editor first startup fails on my openSuse Tumbleweed machine.
Repeating error in the log file:
ImportError: cannot import name 'json' from 'itsdangerous' (/home/ggrau/.local/share/mu/mu_venv-38-20220218-164223/lib/python3.8/site-packages/itsdangerous/__init__.py)
2022-02-18 16:43:24,502 - mu.virtual_environment:643(ensure_and_create) ERROR: Failed to import: flask

on a shell, pip3 list returns:
Flask                             2.0.3
itsdangerous                      2.0.1
"python" version is 2.7.18
"python3" version is 3.8.12
I'm clueless what the installer is trying to do. 
Maybe it should look for specific versions when setting up the virtual environment?
How to figure out what's wrong?

Version: 1.1.0
Locale: C
Platform: Linux 5.16.8-1-default #1 SMP PREEMPT Thu Feb 10 11:31:59 UTC 2022 (5d1f5d2) x86_64

```
Traceback (most recent call last):
  File "/local/projects/MicroBit/mu-master/mu/app.py", line 159, in run
    raise ex
  File "/local/projects/MicroBit/mu-master/mu/app.py", line 146, in run
    venv.ensure_and_create(self.display_text)
  File "/local/projects/MicroBit/mu-master/mu/virtual_environment.py", line 627, in ensure_and_create
    self.ensure()
  File "/local/projects/MicroBit/mu-master/mu/virtual_environment.py", line 667, in ensure
    self.ensure_key_modules()
  File "/local/projects/MicroBit/mu-master/mu/virtual_environment.py", line 750, in ensure_key_modules
    raise VirtualEnvironmentEnsureError(
mu.virtual_environment.VirtualEnvironmentEnsureError: Failed to import: flask
STDERR: Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/ggrau/.local/share/mu/mu_venv-38-20220218-172846/lib/python3.8/site-packages/flask/__init__.py", line 19, in <module>
    from . import json
  File "/home/ggrau/.local/share/mu/mu_venv-38-20220218-172846/lib/python3.8/site-packages/flask/json/__init__.py", line 15, in <module>
    from itsdangerous import json as _json
ImportError: cannot import name 'json' from 'itsdangerous' (/home/ggrau/.local/share/mu/mu_venv-38-20220218-172846/lib/python3.8/site-packages/itsdangerous/__init__.py)

```

**Comment by g3grau** [comment]
Looking closer at the running mu version in the Win VM, it seemed that my issues around flask and itsdangerous are related to a web development module in mu .. which I don't need.

I managed to revert the "add a new mode" instructions from https://mu.readthedocs.io/en/latest/modes.html and removed all remaining relevant parts with the help of grep.
Now mu starts in Linux .. and it works :-) So indeed the problem is only related to the web module.

I guess there is not an easier way to configure mu to run without this web module and flask?

**Comment by woodrow-shen** [comment]
I'm hitting the same issue on Ubuntu 20.04 over pipx too.

**Comment by carlosperate** [comment]
Thanks for the report @g3grau and @woodrow-shen, we've identified the issue and created PR https://github.com/mu-editor/mu/pull/2051 to fix it.

Until we release a new version to PyPI you can probably apply the patch manually and then delete the `mu/wheels/1.1.0b7.zip` to force Mu to download the venv dependencies again.

**Comment by carlosperate** [comment]
A new version with this fix has been released to PyPI 🎉 https://pypi.org/project/mu-editor/1.1.1/

You can be install it via pip and it should now work :)

Thanks for the report and the debugging done!

**Comment by g3grau** [comment]
Thanks @carlosperate for the quick fix! I just started 1.1.1 on Tumbleweed, works like a charm :-)

**Comment by carlosperate** [comment]
Glad that works and thanks for testing it!
